Feature/login page

gabito1451 opened the pull request:
*changed the branch name from gabi to feature/login-page
*changed the value of the select list to accountNumber and text to account Name.
*stored the account Number against the key MB_LOGGEDIN_USER_ACCOUNT_NUMBER in the local storage

*added the logic to redirect the login to transaction page
*changed the function name from storeAcctNo function to login
*removed the header from login and register pages

Gabbiechilz committed:
'added

Gabbiechilz committed:
'added

Gabbiechilz committed:
'login.html'

Gabbiechilz committed:
corrected login-page

nwankwo-ikemefuna commented on index.html:
```diff
@@ -0,0 +1,35 @@
+  <body>
+    <div class="login-session">
+      <h2>Login</h2>
+      <h3>Enter your credentials</h3>
```
This is not necessary. Logging into an account is common knowledge. The user knows what to do. Let's remove it.

nwankwo-ikemefuna commented on index.html:
```diff
@@ -0,0 +1,35 @@
+    <div class="login-session">
+      <h2>Login</h2>
+      <h3>Enter your credentials</h3>
+      <a href="./register.html" id="register-html">Register here</a>
```
The register link is more appropriate at the bottom. I think it will sit better below the "Forgot password" link.

nwankwo-ikemefuna commented on index.html:
```diff
@@ -0,0 +1,35 @@
+      <a href="./register.html" id="register-html">Register here</a>
+      <form onsubmit="return false;" class="login-form">
+        <select id="select" required>
+          <option value="accountNumber" text="accountName">
```
The text should be a meaningful English word i.e. "Account Number" instead of "accountNumber". The text is what the user sees, the value is what the computer understands.

nwankwo-ikemefuna replied:
I believe this should be the default option. The value should be an empty string i.e. `value=""`

nwankwo-ikemefuna commented on index.html:
```diff
@@ -0,0 +1,35 @@
+      </form>
+    </div>
+
+    <script src="./login.js" type="module"></script>
```
Why `type="module"`?

nwankwo-ikemefuna commented on index.html:
```diff
@@ -0,0 +1,35 @@
+      <h3>Enter your credentials</h3>
+      <a href="./register.html" id="register-html">Register here</a>
+      <form onsubmit="return false;" class="login-form">
+        <select id="select" required>
```
Avoid vague names for your identifiers like "select". It's too vague and makes reading your code very difficult. Give it a more descriptive name, such as "account-number".

Secondly, you should use the form component wrapper ("form-group", "form-control", etc) we used on the register page in order to achieve site-wide layout consistency.

nwankwo-ikemefuna commented on index.html:
```diff
@@ -0,0 +1,35 @@
+        <input
+          type="password"
+          placeholder="Pin"
+          id="selected-user-password"
```
Always give your element names that describe what they do to avoid any form of ambiguity. This field collects PIN, not password. A more appropriate name for the id is "account-pin".

nwankwo-ikemefuna commented on login.js:
```diff
@@ -0,0 +1,32 @@
+const passwordElem = document.getElementById("selected-user-password");
```
This should be named something more appropriate and descriptive like `pinElem`. Refer to the comments on index.html file.

nwankwo-ikemefuna commented on login.js:
```diff
@@ -0,0 +1,32 @@
+const passwordElem = document.getElementById("selected-user-password");
+const loginElem = document.querySelector(".login-btn");
+const selectElem = document.getElementById("select");
```
This should be named something more appropriate and descriptive like `accountNumberElem`. Refer to the comments on index.html file.

nwankwo-ikemefuna commented on login.js:
```diff
@@ -0,0 +1,32 @@
+
+let registeredUsers = JSON.parse(localStorage.getItem("MB_USER_ACCOUNTS"));
+for (let i = 0; i < registeredUsers.length; i++) {
+  let optionElem = document.createElement("option");
```
Keyword should be `const` not `let`, since you are not re-assigning the variable.

nwankwo-ikemefuna reviewed (changes requested):
Good attempt 👍 , but needs more work.

nwankwo-ikemefuna commented on login.js:
```diff
@@ -0,0 +1,32 @@
+let registeredUsers = JSON.parse(localStorage.getItem("MB_USER_ACCOUNTS"));
+for (let i = 0; i < registeredUsers.length; i++) {
+  let optionElem = document.createElement("option");
+  let text = document.createTextNode(registeredUsers[i].accountName);
```
Ditto on keyword

nwankwo-ikemefuna replied:
You don't need to create text node here. Change to:
`const text = registeredUsers[i].accountName;`

nwankwo-ikemefuna commented on login.js:
```diff
@@ -0,0 +1,32 @@
+  let text = document.createTextNode(registeredUsers[i].accountName);
+  optionElem.appendChild(text);
+  optionElem.setAttribute("value", registeredUsers[i].accountNumber);
+  selectElem.insertBefore(optionElem, selectElem.lastChild);
```
Suggested:

`selectElem.append(optionElem);`

nwankwo-ikemefuna commented on login.js:
```diff
@@ -0,0 +1,32 @@
+
+// function for logging in
+function login() {
+  localStorage.setItem(
```
No need to `JSON.stringify` here, the `MB_LOGGEDIN_USER_ACCOUNT_NUMBER` stores string. 
You only ever need to stringify arrays and objects.

Suggested:

```
localStorage.setItem(
    "MB_LOGGEDIN_USER_ACCOUNT_NUMBER",
    selectElem.value
  );
```

nwankwo-ikemefuna commented on login.js:
```diff
@@ -0,0 +1,32 @@
+    "MB_LOGGEDIN_USER_ACCOUNT_NUMBER",
+    JSON.stringify(selectElem.value)
+  );
+  if (localStorage.getItem("MB_LOGGEDIN_USER_ACCOUNT_NUMBER") !== null) {
```
This check should be done on visiting the page, not when you click on Login button. Move it to the global scope of this file i.e. move it to the top of the file, outside of this function.

nwankwo-ikemefuna commented on login.js:
```diff
@@ -0,0 +1,32 @@
+    location.href = "transaction.html";
+  }
+  //to check user object with the stored account number
+  let accountUserDetails = registeredUsers.find(
```
Suggested:

```
const accountUserDetails = registeredUsers.find(
    (user) =>
      user.accountNumber ===
      localStorage.getItem("MB_LOGGEDIN_USER_ACCOUNT_NUMBER")
  );
```

Pay attention to the changes I made:
1. Used `const` keyword instead of `let`, because the variable is not being re-assigned.
2. In the `find` method, I renamed each instance of `registeredUsers` from `acctNo` to `user`, because each element (object) in the `registeredUsers` array stores multiple user data (name, account number, pin, etc) and not just account number, so using `acctNo` is misleading.
3. No need to `JSON.parse ` the account number, since it's stored as a string and we need to use it as a string. You only ever need to `JSON.parse` array and objects strings i.e. arrays and objects that must have been converted to strings using `JSON.stringify`.

nwankwo-ikemefuna commented on login.js:
```diff
@@ -0,0 +1,32 @@
+      JSON.parse(localStorage.getItem("MB_LOGGEDIN_USER_ACCOUNT_NUMBER"))
+  );
+  console.log(accountUserDetails);
+}
```
You need to include the logic for redirecting to the Transactions History page upon successful login. 

You can also check if the user exists in our database (although it's necessary in our case but that's a fundamental part of the login action).

Suggested:

```
if (!accountUserDetails) {
  alert('Account not found!');
  return;
}
window.location.href = "transactions.html";
```

Note that the Transactions History page should be named "transactions.html", not "transaction.html", because on it we can see many transactions, not just one. Use the same convention for other page names.

nwankwo-ikemefuna commented on register.html:
```diff
@@ -1,115 +1,112 @@
+    <title>MINI BANK APP</title>
+  </head>
+  <body>
+    <div id="myNav" class="overlay">
```
We shouldn't have this header on the Register page.

nwankwo-ikemefuna commented on register.html:
```diff
@@ -1,115 +1,112 @@
+      </div>
+    </div>
+
+    <!-- <nav>
```
Don't comment out, remove completely. If you think you might need the code later or somewhere else, copy the code to a scratchpad.

nwankwo-ikemefuna commented on register.html:
```diff
@@ -1,115 +1,112 @@
+      <p>Already have an account? <a href="index.html">Sign in</a>.</p>
+    </div>
+
+    <script src="./register.js" type="module"></script>
```
Why `type="module"`?

nwankwo-ikemefuna commented on register.js:
```diff
@@ -1,37 +1,39 @@
-function closeNav() {
-  document.getElementById("myNav").style.width = "0%";
-}
+// function openNav() {
+//   document.getElementById("myNav").style.width = "100%";
+// }
 
+// function closeNav() {
+//   document.getElementById("myNav").style.width = "0%";
+// }
```
Remove commented out code.

nwankwo-ikemefuna commented on styles.css:
```diff
@@ -166,4 +166,51 @@ a {
+  font-size: 16px;
+  color: rgba(0, 0, 0, 38);
+}
+.login-form {
+  width: 100%;
+  max-width: 500px;
+  margin: 0;
+  display: grid;
+  gap: 16px;
+}
+.login-form > a {
+  color: blue;
+}
+.login- form > button {
+  width: 100%;
+  height: 56px;
+  padding: 0 16px;
+  border-radius: 1px;
+  border: 0;
+  font-weight: 600;
+  color: #f9f9f9;
+}
+.login form > input {
+  border: 1px solid #353333;
+  margin-top: 20px;
+}
```
Use the components.css file styles like we used on the Register page in order to achieve consistency.

Gabbiechilz committed:
corrected login page

Gabbiechilz committed:
Merge remote-tracking branch 'origin/main' into feature/login-page

Gabbiechilz committed:
corrected login page

gabito1451 commented:
- removed h3 element from index.html
- placed register link at the bottom of forgot password link
- changed text attribute to Account-Name
- left the value attribute of the option element in the index.html file empty
- gave my id attributes values appropriate names
- deleted commented codes from my work
- removed header element from register file
